### DISCUSSION TITLE
How to work around the unreliable DMChannel.recipient property

### DISCUSSION THREAD

**Discussion by yiays** [nextcord/nextcord#351] [Q&A] [answered]
I'm finding it very frustrating working with DMChannel objects now that

> the recipient information may not be always available.

source: https://nextcord.readthedocs.io/en/latest/api.html#nextcord.DMChannel.recipient

I don't understand why this has changed, I assume it's the fault of the API somehow, it's very difficult to interact with users in DMs if I don't know who they are. Of course I can determine who they are when responding to a message or command, but not when I'm sending a message to a stored DMChannel id.

Could someone enlighten me as to why and when recipient information will be missing from a DMChannel object?

**Comment by ooliver1** [comment, marked as answer]
i know this is very late but im working through the discussions, https://discord.com/developers/docs/resources/channel#channel-object is the discord docs for this

no idea why the docs would say this, however it seems discord may not send this `recipients` field or something, so assume it will be available and if it isnt then you can followup at https://discord.gg/discord-developers and ask them why this is
